testing/src/cypress/e2e/02-institution/04-manage-location.cy.ts

```diff
@@ -1,121 +1,518 @@
+    data.validData.stringsWithAlphaNumericSpecialChars.forEach(
+      (text: string) => {
+        institutionManageLocationObject.lastNameInputText().clear().type(text);
+        cy.contains("Last name must ").should("not.exist");
+      }
+    );
+  });
+
+  it("Verify that LastName should not accept more than 100 chars", () => {
+    data.invalidData.stringWithMoreThan100Chars.forEach((text: string) => {
+      institutionManageLocationObject.lastNameInputText().clear().type(text);
+      cy.contains("Last name must ").should("not.exist");
+    });
+  });
+
+  it("Verify that email field does not accept invalid emails", () => {
+    verifyInvalidEmailFormatsAreNotAllowed();
+  });
+
+  it("Verify that email field accepts valid emails", () => {
+    verifyValidEmailFormatsAreAllowed();
+  });
+
+  it("Verify that phone number field does not accept numbers below 10 and above 20", () => {
+    verifyPhoneNumberFieldNotAcceptNumBelow10Above20();
+  });
+
+  it("Verify that phone number field accepts valid numbers", () => {
+    verifyPhoneNumberFieldAcceptNumBetween10And20();
+  });
+});
+
+describe("[Institution Manage Location] - Add New Location and update for the institution", () => {
+  const uniqeId1 = institutionHelperActions.getUniqueId();
+  const name1 = `test-${uniqeId1}`;
+  const institutionCode = "ASKN";
+  const canadaPostalCode = "A1A2A3";
+  const phoneNumber = "1236549871";
+
+  before(() => {
+    loginAndClickOnAddNewLocation();
+  });
+
+  it("Create new institution location", () => {
+    institutionManageLocationObject
+      .locationName()
       .clear()
-      .type(manageLocation.firstName);
+      .type(`location-${name1}`)
+      .should("have.value", `location-${name1}`);
     institutionManageLocationObject
-      .lastNameINputText()
+      .institutionCode()
       .clear()
-      .type(manageLocation.lastName);
+      .type(institutionCode)
+      .should("have.value", institutionCode);
     institutionManageLocationObject
-      .emailInputText()
+      .addressLine1()
       .clear()
-      .type(manageLocation.email);
-    cy.focused();
+      .type(`Addrline1-${name1}`)
+      .should("have.value", `Addrline1-${name1}`);
     institutionManageLocationObject
-      .phoneInputText()
+      .addressLine2()
       .clear()
-      .type(manageLocation.phoneNumber);
+      .type(`Addrline2-${name1}`)
+      .should("have.value", `Addrline2-${name1}`);
+    institutionManageLocationObject.countryDropDownMenu().click();
+    institutionManageLocationObject.countryCanadaFromDropDownMenu().click();
+    institutionManageLocationObject.provinceDropDownMenu().click();
+    institutionManageLocationObject.provinceABFromDropDown().click();
+    institutionManageLocationObject
+      .cityInputText()
+      .type(`city-${name1}`)
+      .should("have.value", `city-${name1}`);
+    institutionManageLocationObject
+      .canadaPostalCode()
+      .type(canadaPostalCode)
+      .should("have.value", "A1A 2A3");
```

Comment:
I see postcode is already defined above ![image](https://user-images.githubusercontent.com/77353155/190245085-d0807894-b85c-4191-addc-86416e93d628.png)

Reply:
addressed during the call